splink/cluster_metrics.py

```diff
@@ -0,0 +1,59 @@
+def _size_density_sql(
+    df_predict, df_clustered, threshold_match_probability, _unique_row_id
+):
+    """Generates sql for computing cluster size and density at a given threshold.
+
+    Args:
+        df_predict (SplinkDataFrame): The results of `linker.predict()`
+        df_clustered (SplinkDataFrame): The outputs of
+                `linker.cluster_pairwise_predictions_at_threshold()`
+        threshold_match_probability (float): Filter the pairwise match
+            predictions to include only pairwise comparisons with a
+            match_probability above this threshold.
+        _unique_row_id (string): name of unique id column in settings dict
+
+    Returns:
+        sql string for computing cluster size and density
+    """
+
+    # Get physical table names from Splink dataframes
+    edges_table = df_predict.physical_name
+    clusters_table = df_clustered.physical_name
+
+    sqls = []
+    sql = f"""
+        SELECT
+            {_unique_row_id}_l,
+            COUNT(*) AS count_edges
+        FROM {edges_table}
+        WHERE match_probability >= {threshold_match_probability}
+        GROUP BY {_unique_row_id}_l
+    """
+
+    sql = {"sql": sql, "output_table_name": "__count_edges"}
```

Comment:
Not hugely important, and certainly not done consistently throughout the codebase, but the advantage of having all tables as `__splink__X` (even ones that generally only end up in CTEs) is that there is potentially less pollution in persistent backends if pipelining is switched off (i.e. in debug mode)